src/components/site/tableOfContents/index.jsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import reactn from 'reactn';
+import { Link, graphql, StaticQuery } from 'gatsby';
+import PropTypes from 'prop-types';
+import { Drawer } from 'react-md';
+import styles from './table-of-contents.module.scss';
+
+// import Footer from '../components/site/footer';
+const TO_PREFIX = '/';
+const routes = [
+  {
+    label: 'home',
+    to: `${TO_PREFIX}`,
+    icon: 'home',
+    exact: 'true',
+    primaryText: 'Home',
+    active: false,
+  },
+  {
+    label: 'style-guide',
+    to: `${TO_PREFIX}StyleGuide`,
+    icon: 'style',
+    primaryText: 'Style Guide',
+    active: false,
+  },
+  { divider: true },
+  {
+    primaryText: 'Table of Contents',
+    subheader: true,
+  },
+];
+
+@reactn
+class TableOfContents extends React.PureComponent {
+  // constructor(props) {
+  //   super(props);
+  //
+  // }
+
+  componentDidMount() {
+    const { navLinks } = this.props;
+    const links = routes.concat(navLinks);
+    this.navLinks = links.map(link => {
+      if (link.divider || link.subheader) return link;
+
+      if (link.to) {
+        link.component = Link;
+      }
+
+      if (link.slug) {
+        link.component = Link;
+        link.label = link.title;
+        link.to = `${TO_PREFIX}${link.slug}`;
+        link.primaryText = link.title;
+        link.active = false;
+      }
+      return link;
+    });
+  }
```

Comment:
Instead of creating, and using/subsequently mutating the Class variable `this.navLinks`, the more Reacty way to accomplish this is actually to encapsulate all of this routes/navLinks logic into a class method that returns the navLinks, and then call that method  when you're setting the `navItems` prop on line 81.  This will actually also set you up to be updating that active state stuff in one place as well (since the render method runs on both mount and update).

Reply:
I can almost visualize what you're referring to. I'll have to wrap my mind around the Reacty way of doing the stuffs. I want to and I am going to use this method. Just need to clearly visualize it first.

Reply:
Here is how I resolved this:

```
  componentDidMount() {
    this.updateNavLinks();
  }

  componentDidUpdate() {
    this.updateNavLinks();
  }

  updateNavLinks = () => {
    const { navLinks } = this.props;
    const massagedNavLinks = this.getNavLinks(navLinks);
    this.navLinks = [...this.routes, ...massagedNavLinks];
  };

  setActivePage = linkId => {
    return linkId === this.global.pageId;
  };

  getNavLinks = navLinks => {
    return navLinks.map(link => {
      if (link.divider || link.subheader) return link;
      return {
        ...link,
        component: Link,
        label: link.title,
        to: `${this.TO_PREFIX}${link.slug}`,
        primaryText: link.title,
        active: this.setActivePage(link.id),
      };
    });
  };
```